extensions-core/stats/src/main/java/io/druid/query/aggregation/variance/VarianceAggregatorFactory.java

```diff
@@ -0,0 +1,266 @@
+{
+  private static final byte CACHE_TYPE_ID = 16;
+
+  private final String fieldName;
+  private final String name;
+  private final boolean variancePop;
+
+  @JsonCreator
+  public VarianceAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") String fieldName,
+      @JsonProperty("variancePop") boolean variancePop
+  )
+  {
+    Preconditions.checkNotNull(name, "Must have a valid, non-null aggregator name");
+    Preconditions.checkNotNull(fieldName, "Must have a valid, non-null fieldName");
+
+    this.name = name;
+    this.fieldName = fieldName;
+    this.variancePop = variancePop;
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return "variance";
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return VarianceHolder.getMaxIntermediateSize();
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    ValueType valueType = metricFactory.columnType(fieldName);
+    if (valueType == null) {
+      return Aggregators.noopAggregator();
+    }
+    if (valueType == ValueType.FLOAT) {
+      return new VarianceAggregator.FloatVarianceAggregator(
+          name,
+          metricFactory.makeFloatColumnSelector(fieldName)
+      );
+    }
+    if (valueType == ValueType.LONG) {
+      return new VarianceAggregator.LongVarianceAggregator(
+          name,
+          metricFactory.makeLongColumnSelector(fieldName)
+      );
+    }
+    return new VarianceAggregator.ObjectVarianceAggregator(
+        name,
+        metricFactory.makeObjectColumnSelector(fieldName)
+    );
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    ValueType valueType = metricFactory.columnType(fieldName);
+    if (valueType == null) {
+      return Aggregators.noopBufferAggregator();
+    }
+    if (valueType == ValueType.FLOAT) {
+      return new VarianceBufferAggregator.FloatVarianceAggregator(
+          name,
+          metricFactory.makeFloatColumnSelector(fieldName)
+      );
+    }
+    if (valueType == ValueType.LONG) {
+      return new VarianceBufferAggregator.LongVarianceAggregator(
+          name,
+          metricFactory.makeLongColumnSelector(fieldName)
+      );
+    }
+    return new VarianceBufferAggregator.ObjectVarianceAggregator(
+        name,
+        metricFactory.makeObjectColumnSelector(fieldName)
+    );
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new VarianceAggregatorFactory(name, name, variancePop);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Arrays.<AggregatorFactory>asList(new VarianceAggregatorFactory(fieldName, fieldName, variancePop));
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass()) {
```

Comment:
suggest `this.getName().equals(other.getName()))` since `other.getName()` *probably* returns a value, but `this.getName()` is guaranteed in the constructor to return a non-null value.